### ISSUE TITLE
Essentials v20 sqlite

### ISSUE THREAD

**Issue by makarasty** [Kieaer/Essentials#132]
the error occurs after restarting the server, **and is solved only by deleting database.db**
the plugin configuration is as basic as possible without any changes
tested on java 17 and 21
**Logs**
```
C:\Users\MaKarastY\Desktop\Test\mintest>set JAVA_HOME = C:\Java\jdk-17\bin
C:\Users\MaKarastY\Desktop\Test\mintest>java -Xms128M -Xmx1024M -jar server-release.jar
[12-09-2024 23:12:15] [I] 6 mods loaded.
[12-09-2024 23:12:15] [I] Downloading postgresql-42.7.3.jar...
[12-09-2024 23:12:15] [I] postgresql-42.7.3.jar saved
[12-09-2024 23:12:15] [I] Downloading mysql-connector-j-8.4.0.jar...
[12-09-2024 23:12:16] [I] mysql-connector-j-8.4.0.jar saved
[12-09-2024 23:12:16] [I] Downloading mariadb-java-client-3.4.0.jar...
[12-09-2024 23:12:16] [I] mariadb-java-client-3.4.0.jar saved
[12-09-2024 23:12:16] [I] Downloading sqlite-jdbc-3.46.0.0.jar...
[12-09-2024 23:12:17] [I] sqlite-jdbc-3.46.0.0.jar saved
[12-09-2024 23:12:17] [I] Downloading h2-2.2.224.jar...
[12-09-2024 23:12:18] [I] h2-2.2.224.jar saved
[12-09-2024 23:12:18] [I] Downloading mssql-jdbc-12.6.2.jre11.jar...
[12-09-2024 23:12:18] [I] mssql-jdbc-12.6.2.jre11.jar saved
[12-09-2024 23:12:19] [I] [Essential] The current plugin version is the latest version!
[12-09-2024 23:12:19] [I] [Essential] Loaded.
[12-09-2024 23:12:19] [I] Server loaded. Type 'help' for help.
[12-09-2024 23:12:19] [I] Config file updated!
[12-09-2024 23:12:19] [I] Config file updated!
[12-09-2024 23:12:19] [I] Config file updated!
host
[12-09-2024 23:12:34] [I] Randomized next map to be Labyrinth.
[12-09-2024 23:12:34] [I] Loading map...
[12-09-2024 23:12:34] [I] Map loaded.
[12-09-2024 23:12:34] [I] Opened a server on port 6567.
[12-09-2024 23:12:47] [I] ? MaKarastY has connected. [pU6gLelVSFQAAAAAJlBw9w==]
^CTerminate batch job (Y/N)? n



C:\Users\MaKarastY\Desktop\Test\mintest>goto l
C:\Users\MaKarastY\Desktop\Test\mintest>set JAVA_HOME = C:\Java\jdk-17\bin
C:\Users\MaKarastY\Desktop\Test\mintest>java -Xms128M -Xmx1024M -jar server-release.jar
[12-09-2024 23:12:54] [I] 6 mods loaded.
[12-09-2024 23:12:55] [E] java.lang.RuntimeException: Error loading mod essential
        at mindustry.mod.Mods.contextRun(Mods.java:797)
        at mindustry.mod.Mods.lambda$eachClass$30(Mods.java:785)
        at arc.struct.Seq.each(Seq.java:182)
        at mindustry.mod.Mods.eachClass(Mods.java:785)
        at mindustry.server.ServerLauncher.init(ServerLauncher.java:80)
        at arc.backend.headless.HeadlessApplication.mainLoop(HeadlessApplication.java:64)
        at arc.backend.headless.HeadlessApplication$1.run(HeadlessApplication.java:52)
Caused by: org.jetbrains.exposed.exceptions.ExposedSQLException: java.sql.SQLException: The prepared statement has been finalized
SQL: []
        at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed_core(Statement.kt:83)
        at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:292)
        at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:269)
        at org.jetbrains.exposed.sql.statements.Statement.execute(Statement.kt:59)
        at org.jetbrains.exposed.sql.statements.jdbc.JdbcConnectionImpl.executeInBatch(JdbcConnectionImpl.kt:130)
        at org.jetbrains.exposed.sql.Transaction.execInBatch(Transaction.kt:276)
        at org.jetbrains.exposed.sql.SchemaUtils.execStatements(SchemaUtils.kt:362)
        at org.jetbrains.exposed.sql.SchemaUtils.create(SchemaUtils.kt:373)
        at essential.core.DB.createTable$lambda$5(DB.kt:120)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:324)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.access$inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:1)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$inTopLevelTransaction$1.invoke(ThreadLocalTransactionManager.kt:371)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:379)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction(ThreadLocalTransactionManager.kt:370)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$transaction$1.invoke(ThreadLocalTransactionManager.kt:279)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:379)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:249)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:227)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$default(ThreadLocalTransactionManager.kt:226)
        at essential.core.DB.createTable(DB.kt:119)
        at essential.core.Main.init(Main.kt:103)
        at mindustry.mod.Mods.lambda$eachClass$29(Mods.java:785)
        at mindustry.mod.Mods.contextRun(Mods.java:795)
        ... 6 more
Caused by: java.sql.SQLException: The prepared statement has been finalized
        at org.sqlite.core.NativeDB.throwex(NativeDB.java:500)
        at org.sqlite.core.NativeDB.column_count(Native Method)
        at org.sqlite.core.DB.column_names(DB.java:862)
        at org.sqlite.core.SafeStmtPtr.safeRun(SafeStmtPtr.java:128)
        at org.sqlite.core.CorePreparedStatement.<init>(CorePreparedStatement.java:47)
        at org.sqlite.jdbc3.JDBC3PreparedStatement.<init>(JDBC3PreparedStatement.java:32)
        at org.sqlite.jdbc4.JDBC4PreparedStatement.<init>(JDBC4PreparedStatement.java:25)
        at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:34)
        at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:225)
        at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:205)
        at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:210)
        at org.jetbrains.exposed.sql.statements.jdbc.JdbcConnectionImpl.prepareStatement(JdbcConnectionImpl.kt:80)
        at org.jetbrains.exposed.sql.statements.Statement.prepared(Statement.kt:46)
        at org.jetbrains.exposed.sql.statements.jdbc.JdbcConnectionImpl$executeInBatch$prepStatement$1.prepared(JdbcConnectionImpl.kt:102)
        at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed_core(Statement.kt:79)
        ... 28 more
```

**Additional context (MODS)**
```
│   Essential.jar
│   EssentialAchievements.jar
│   EssentialBridge.jar
│   EssentialChat.jar
│   EssentialDiscord.jar
│   EssentialProtect.jar
│
└───Essentials
    │   chat_blacklist.txt
    │   permission.yaml
    │   permission_user.yaml
    │
    ├───config
    │       config.yaml
    │       config_bridge.yaml
    │       config_chat.yaml
    │       config_discord.yaml
    │       config_protect.yaml
    │
    ├───data
    │       database.db
    │       ipv4.txt
    │
    ├───drivers
    │       h2-2.2.224.jar
    │       mariadb-java-client-3.4.0.jar
    │       mssql-jdbc-12.6.2.jre11.jar
    │       mysql-connector-j-8.4.0.jar
    │       postgresql-42.7.3.jar
    │       sqlite-jdbc-3.46.0.0.jar
    │
    └───log
        │   Player.log
        │   Tap.log
        │
        └───old
            ├───Player
            └───Tap
 ```

**Comment by Kieaer** [comment]
It seems SQLite JDBC [bug](https://github.com/xerial/sqlite-jdbc/issues/731) or database corrupted.
To improve performance, I don't save to the DB while the player is on the server, so a force quit will result in loss of player data.

You must use ``exit`` command to save all data to database.

**Comment by makarasty** [comment]
I understand the importance of using the `exit` command to shut down the server. However, I would like to see the plugin work stably with SQLite. Maybe it's worth considering using, for example, [HikariCP](https://github.com/brettwooldridge/HikariCP) (not sure)

**Comment by Kieaer** [comment]
I've been using SQLite on my server for 3 months with no issues.

**Comment by makarasty** [comment]
I'm glad to hear it's working well for you. However, I believe the plugin should be robust enough to handle various non-standard situations to ensure stability for all users

**Comment by makarasty** [comment]
If you need additional information, I can provide it.

**Comment by makarasty** [comment]
Most likely fixed in #136
